chain-extensions/pallet-assets/src/tests.rs

```diff
@@ -0,0 +1,478 @@
+        });
+}
+
+#[test]
+fn getters_works() {
+    ExtBuilder::default()
+        .existential_deposit(50)
+        .build()
+        .execute_with(|| {
+            let addr = instantiate();
+
+            // Arrange
+            // Alice creates & mint token
+            assert_ok!(Assets::create(
+                RuntimeOrigin::signed(ALICE),
+                ASSET_ID,
+                ALICE,
+                1
+            ));
+            assert_ok!(Assets::mint(
+                RuntimeOrigin::signed(ALICE),
+                ASSET_ID,
+                ALICE,
+                1000
+            ));
+            assert_ok!(Assets::approve_transfer(
+                RuntimeOrigin::signed(ALICE),
+                ASSET_ID,
+                BOB,
+                100
+            ));
+            assert_ok!(Assets::set_metadata(
+                RuntimeOrigin::signed(ALICE),
+                ASSET_ID,
+                "Token".as_bytes().to_vec(),
+                "TKN".as_bytes().to_vec(),
+                1
+            ));
+
+            // Assert - verify state using chain extension getters
+            assert_eq!(
+                allowance(addr.clone(), ASSET_ID, ALICE, BOB).data[1..],
+                100u128.encode()
+            );
+            assert_eq!(
+                balance_of(addr.clone(), ASSET_ID, ALICE).data[1..],
+                1000u128.encode()
+            );
+            assert_eq!(
+                total_supply(addr.clone(), ASSET_ID).data[1..],
+                1000u128.encode()
+            );
+            assert_eq!(metadata_decimals(addr.clone(), ASSET_ID).data[1..], [1u8]);
+            assert_eq!(
+                metadata_name(addr.clone(), ASSET_ID).data[1..],
+                "Token".encode()
+            );
+            assert_eq!(
+                metadata_symbol(addr.clone(), ASSET_ID).data[1..],
+                "TKN".encode()
+            );
+            assert_eq!(
+                minimum_balance(addr.clone(), ASSET_ID).data[1..],
+                1u128.encode()
+            );
+        });
+}
+
+fn instantiate() -> AccountId32 {
+    let code = fs::read("../../tests/ink-contracts/pallet_assets_extension.wasm")
+        .expect("could not read .wasm file");
+    let _ = Balances::deposit_creating(&ALICE, ONE * 1000);
+    let _ = Balances::deposit_creating(&BOB, ONE * 1000);
+    let instance_selector: Vec<u8> = selector_bytes!("new").to_vec();
+    Contracts::bare_instantiate(
+        ALICE,
+        0,
+        GAS_LIMIT,
+        Some(ONE),
+        Code::Upload(code),
+        instance_selector,
+        vec![],
+        DebugInfo::Skip,
+        CollectEvents::UnsafeCollect,
+    )
+    .result
+    .unwrap()
+    .account_id
+}
+
+fn transfer(
+    addr: AccountId32,
+    asset_id: u128,
+    target: AccountId32,
+    amount: u128,
+) -> Result<ExecReturnValue, DispatchError> {
+    let data = [
+        selector_bytes!("transfer").to_vec(),
```

Comment:
Nitpick - these strings could have been left as &str consts.

Reply:
You mean like `const TRASNFER &str = b"9bae9d5e";` ? 
I think this way is more readable (even tho it's not the most performant - but it's off chain tests)

Reply:
Almost, I meant like `const TRANSFER: &str = "transfer";

Reply:
Yeah but I would have to hash that  &str and get the first 4 bytes so it defeats the static part